main.cpp

```diff
@@ -1,10 +1,27 @@
+#include "main.h"
+
 #include <iostream>
+#include <string>
 
 #include <GL/glew.h>
 #include <GLFW/glfw3.h>
 #include <glm/glm.hpp>
 #include <glm/gtc/matrix_transform.hpp>
 
+#include "indexbuffer.h"
+#include "rectangle.h"
+#include "renderer.h"
+#include "renderutils.h"
+#include "shader.h"
+#include "texture.h"
+#include "vertexarray.h"
+#include "vertexbuffer.h"
+#include "vertexbufferlayout.h"
+
+Rectangle* player = nullptr;
```

Comment:
This is a temporary thing while scope things better into more functions and classes. Player is declared here for use in the KeyCallback() and in main(), but needs to wait to be initialized after opengl is properly set up. Otherwise many issues will occur.